### PR TITLE
Vm detail actions

### PR DESCRIPTION
**Pull request by mareklibra**

### CODE REVIEW

File: src/components/VmActions/index.js [hunk 1]
```diff
@@ -89,7 +86,22 @@ const VmActions = ({ vm, actions, isOnCard = false }) => {
...
+  onStart: PropTypes.func.isRequired,
+  onSuspend: PropTypes.func.isRequired,
 }
 
-export default VmActions
+export default connect(
+  (state) => ({
+    icons: state.icons,
+  }),
+  (dispatch, { vm }) => ({
```
Comment (matobet) [inline src/components/VmActions/index.js:100]: Nice! I really like this destructuring of the `ownProps`!

File: src/components/VmDetail/index.js [hunk 2]
```diff
@@ -73,7 +75,14 @@ class VmDetail extends Component {
...
-  actions: PropTypes.object.isRequired,
+  onCloseVmDetail: PropTypes.func.isRequired,
 }
 
-export default VmDetail
+export default connect(
+  (state) => ({
+    icons: state.icons,
+  }),
+  (dispatch, ownProps) => ({
```
Comment (matobet) [inline src/components/VmDetail/index.js]: no need to take the `ownProps` argument if it is not used a plain `(dispatch) => ({ ...` is sufficient

File: src/components/VmsList/Vm.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+  vm: PropTypes.object.isRequired,
+  icons: PropTypes.object.isRequired,
+  onSelectVm: PropTypes.func.isRequired,
+}
+
+export default connect(
+  (state) => ({
+    icons: state.icons,
+  }),
+  (dispatch, ownProps) => ({
```
Comment (matobet) [inline src/components/VmsList/Vm.js]: here you can also use the destructuring of `ownProps` as in `(dispatch, { vm }) => ...`

File: src/components/VmsList/VmStatusText.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+  }
+
+  switch (status) { // TODO: review VM states
+    case 'up':
+    case 'powering_up':
+    case 'paused':
+    case 'migrating':
+    default:
+      return (
+        <p className='card-pf-info text-center' >
```
Comment (matobet) [inline src/components/VmsList/VmStatusText.js]: possible extra space between the final `'` and `>`

File: src/components/VmsList/Vms.js [hunk 5]
```diff
@@ -0,0 +1,32 @@
+import React, { PropTypes } from 'react'
+import { connect } from 'react-redux'
+
+import style from './style.css'
+import Vm from './Vm'
+
+const Vms = ({ vms }) => {
+  const selectedVmId = vms.get('selected')
+  const containerClass = 'container-fluid container-cards-pf ' + (selectedVmId ? style['move-left'] : style['move-left-remove'])
```
Comment (matobet) [inline src/components/VmsList/Vms.js:9]: note to self: we should really look into using https://github.com/JedWatson/classnames for such computed styles

File: src/components/VmsList/Vms.js [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+}
+Vms.propTypes = {
+  vms: PropTypes.object.isRequired,
+}
+
+export default connect(
+  (state) => ({
+    vms: state.vms,
+  }),
+  (dispatch) => ({ })
```
Comment (matobet) [inline src/components/VmsList/Vms.js]: no need to list the `mapDispatchToProps` second argument if it is not used

File: src/components/VmsPageHeader/LoginMenu.js [hunk 7]
```diff
@@ -0,0 +1,50 @@
...
+import style from './style.css'
+
+const LoginMenu = ({ config, onLogout, onLogin }) => {
+  if (config.get('loginToken')) {
+    return (
+      <li className='dropdown'>
+        <a className={'dropdown-toggle ' + style['user-name']} data-toggle='dropdown' href='#'>
+          <i className='fa fa-sign-out' aria-hidden='true' />&nbsp;
+          {config.getIn(['user', 'name'])}
+          <b className='caret' />
```
Comment (matobet) [inline src/components/VmsPageHeader/LoginMenu.js:15]: question: why are we applying a style on an empty `b` element?

**Review by matobet — CHANGES_REQUESTED** [review]

File: src/components/VmsPageHeader/LoginMenu.js (hunk 7 above)
Comment (mareklibra) [reply src/components/VmsPageHeader/LoginMenu.js:15]: since it renders the caret symbol (the 'down arrow')

**Comment by mareklibra** [comment]
Updated for the requested changes

**Review by matobet — APPROVED** [review]

**Pull request merged**